jwtmiddleware.go

```diff
@@ -26,95 +61,132 @@ type TokenExtractor func(r *http.Request) (string, error)
 // happen. In the default implementation we can add safe defaults for those.
 type ValidateToken func(string) (interface{}, error)
 
-// Options is a struct for specifying configuration options for the middleware.
-type Options struct {
-	// Validate handles validating a token.
-	Validate ValidateToken
+type JWTMiddleware struct {
+	// validateToken handles validating a token.
+	validateToken ValidateToken
 	// The name of the property in the request where the user information
 	// from the JWT will be stored.
 	// Default value: "user"
-	UserProperty string
+	contextKey string
 	// The function that will be called when there are errors in the
 	// middleware.
 	// Default value: OnError
-	ErrorHandler errorHandler
+	errorHandler ErrorHandler
 	// A boolean indicating if the credentials are required or not
 	// Default value: false
-	CredentialsOptional bool
+	credentialsOptional bool
 	// A function that extracts the token from the request
 	// Default: FromAuthHeader (i.e., from Authorization header as bearer token)
-	Extractor TokenExtractor
+	tokenExtractor TokenExtractor
 	// Debug flag turns on debugging output
 	// Default: false
-	Debug bool
-	// When set, all requests with the OPTIONS method will use authentication
-	// Default: false
-	EnableAuthOnOptions bool
+	debug bool
+	// When set, all requests with the OPTIONS method will have their
+	// token validated.
+	// Default: true
+	validateOnOptions bool
 }
 
-type JWTMiddleware struct {
-	Options Options
+// Option is how options for the middleware are setup.
+type Option func(*JWTMiddleware)
```

Comment:
I opted to go with the options pattern as that is what we are doing in the josev2 validation piece.